gamemodes/terrortown/entities/entities/ttt_beacon.lua

```diff
@@ -101,6 +101,16 @@ function ENT:Think()
 
 			if not IsValid(ent) or not ent:IsPlayer() then continue end
 
+			---
+			-- Hook that is called when a player is about to be found by a beacon.
+			-- With this hook the player can be ignored.
+			-- @param Player ply The player that the beacon has found
+			-- @param Entity ent The beacon entity that found the player
+			-- @return boolean Return false to cancel the player being detected
+			-- @hook
+			-- @realm server
+			if hook.run("TTT2BeaconDetectPlayer", self, ent) == false then continue end
```

Comment:
nope it should be put somewhere else in this file.

It shoould look something like this:
```lua
if SERVER then
	-- your doc here
	function GAMEMODE:TTT2BeaconDetectPlayer(ply, ent) end
end
```

Reply:
it is `GAMEMODE` here because when entities are loaded `GM` is no longer valid and `GAMEMODE` has to be used instead

Reply:
Think I've done it.